app/helpers/application_helper.rb

```diff
@@ -36,9 +36,11 @@ def current_page
   # @return [TrueClass] true when we should render the given link.
   # @return [FalseClass] false when we should **not** render the given link.
   def display_navigation_link?(link:)
-    # This is a quick short-circuit; we already have the link.  So don't bother asking the "Is this
-    # feature enabled" question if the given link requires a sign in and the user is not signed in.
-    return false if link.display_only_when_signed_in? && !user_signed_in?
+    # This is a quick short-circuit; we already have the link. So don't bother asking the "Is this
+    # feature enabled" question if the given link requires a logged in/out state
+    # that doesn't match the user's current state.
+    return false if link.display_to == "logged_in" && !user_signed_in?
+    return false if link.display_to == "logged_out" && user_signed_in?
```

Comment:
With `_prefix: true` the `display_to` is prepended in the helper methods, which allows us to use Rails' ~~magic~~ autogenerated `enum` helper methods like so:

```suggestion
    return false if link.display_to_logged_in? && !user_signed_in?
    return false if link.display_to_logged_out? && user_signed_in?
```